src/yugecin/opsudance/Dancer.java

```diff
@@ -175,27 +197,99 @@ public void update(int time, GameObject p, GameObject c) {
 				double[] spinnerStartPoint = spinner.getPoint();
 				c.start = new Vec2f((float) spinnerStartPoint[0], (float) spinnerStartPoint[1]);
 			}
-			if (p == d) {
-				mover = new LinearMover(p, c, dir);
-			} else {
-				mover = moverFactory.create(p, c, dir);
-			}
+			if (mover == null || mover.getEnd() != c)
```

Comment:
Reusing the mover makes it a bit more effizient, because some movers (mine especially) need some calculations when being initialized